iaso/api/microplanning.py

```diff
@@ -313,10 +331,11 @@ def validate(self, attrs):
         if team.project != project:
             validation_errors["team"] = "planningAndTeams"
 
-        forms = validated_data.get("forms", self.instance.forms if self.instance else None)
-        for form in forms:
-            if form not in project.forms.all():
-                validation_errors["forms"] = "planningAndForms"
+        forms = validated_data.get("forms", list(self.instance.forms.all()) if self.instance else None)
+        if forms:
+            for form in forms:
+                if form not in project.forms.all():
+                    validation_errors["forms"] = "planningAndForms"
```

Comment:
You are looping even after the error is found. You can replace this with this code:
```suggestion
        project_forms = project.forms.all()
        if forms and not all(f in project_forms for f in forms):
             validation_errors["forms"] = "planningAndForms"
```